native/cocos/bindings/jswrapper/v8/Object.cpp

```diff
@@ -51,6 +51,55 @@ uint32_t nativeObjectId = 0;
     #endif
 } // namespace
 
+class JSBPersistentHandleVisitor : public v8::PersistentHandleVisitor {
+public:
+    JSBPersistentHandleVisitor() = default;
+
+    void VisitPersistentHandle(v8::Persistent<v8::Value> *value, uint16_t classId) override {
+        if (value == nullptr || classId != ObjectWrap::MAGIC_CLASS_ID_JSB) {
+            return;
+        }
+
+        auto &persistObj = v8::Persistent<v8::Object>::Cast(*value);
+        const int fieldCount = v8::Object::InternalFieldCount(persistObj);
+        if (fieldCount != 1) {
+            return;
+        }
+
+        void *ptr = v8::Object::GetAlignedPointerFromInternalField(persistObj, 0);
+        if (ptr == nullptr) {
+            return;
+        }
+
+        auto *obj = reinterpret_cast<Object *>(ptr);
+        auto *nativeObj = obj->getPrivateData();
+        if (nativeObj == nullptr) {
+            // Not a JSB binding object
+            return;
+        }
+
+        // Remove mapping
+        auto iter = se::NativePtrToObjectMap::find(nativeObj);
+        if (iter != se::NativePtrToObjectMap::end()) {
+            se::NativePtrToObjectMap::erase(iter);
+        }
```

Comment:
Call erase directly?

Reply:
Yes, could do that, but use `find` and `erase` is more explicit.